src/tests/unit/test_live_response_api.py

```diff
@@ -384,6 +469,111 @@ def test_get_file(cbcsdk_mock, connection_mock):
+                             GET_FILE_COMMAND_RESP)
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/liveresponse/sessions/1:2468/commands/7',
+                             GET_FILE_END_RESP)
+    cbcsdk_mock.mock_request('DELETE', '/appservices/v6/orgs/test/liveresponse/sessions/1:2468', None)
+    manager = LiveResponseSessionManager(cbcsdk_mock.api)
+    with manager.request_session(2468) as session:
+        command_id, _ = session.get_raw_file('c:\\\\test.txt', async_mode=True)
+        assert command_id == 7
+
+
+def test_command_status(cbcsdk_mock):
+    """Test command status method"""
+    cbcsdk_mock.mock_request('POST', '/appservices/v6/orgs/test/liveresponse/sessions', SESSION_INIT_RESP)
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/liveresponse/sessions/1:2468', SESSION_POLL_RESP)
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/devices/2468', DEVICE_RESPONSE)
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/liveresponse/sessions/1:2468/commands/7',
+                             GET_FILE_END_RESP)
+    cbcsdk_mock.mock_request('DELETE', '/appservices/v6/orgs/test/liveresponse/sessions/1:2468', None)
+    manager = LiveResponseSessionManager(cbcsdk_mock.api)
+    with manager.request_session(2468) as session:
+        status = session.command_status(7)
+        assert status == 'COMPLETE'
+
+
+def test_session_status(cbcsdk_mock):
+    """Test command status method"""
+    cbcsdk_mock.mock_request('POST', '/appservices/v6/orgs/test/liveresponse/sessions', SESSION_INIT_RESP)
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/liveresponse/sessions/1:2468', SESSION_POLL_RESP)
+    manager = LiveResponseSessionManager(cbcsdk_mock.api)
+    session_id, _ = manager.request_session(2468, async_mode=True)
+    status = manager.session_status(session_id)
+    assert status == 'ACTIVE'
+
+
+def test_cancel_complete_command(cbcsdk_mock):
+    """Test the response to the 'cancel command' command for completed command."""
+    cbcsdk_mock.mock_request('POST', '/appservices/v6/orgs/test/liveresponse/sessions', SESSION_INIT_RESP)
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/liveresponse/sessions/1:2468', SESSION_POLL_RESP)
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/devices/2468', DEVICE_RESPONSE)
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/liveresponse/sessions/1:2468/commands/7',
+                             GET_FILE_END_RESP)
+    cbcsdk_mock.mock_request('DELETE', '/appservices/v6/orgs/test/liveresponse/sessions/1:2468', None)
+    manager = LiveResponseSessionManager(cbcsdk_mock.api)
+    with manager.request_session(2468) as session:
+        with pytest.raises(ApiError) as excinfo:
+            session.cancel_command(7)
+        assert excinfo.value.__str__().startswith('Cannot cancel command in status COMPLETE')
+
+
+def test_cancel_pending_command(cbcsdk_mock):
```

Comment:
We need a test to verify the logic flow for when a user cancels a command and what happens if a user then does a `command.results()`. will quickly return None? or some other result? or will it hang?

Do we also have a sample if an exception is raised during the async loop? How does the logic differ for each location we run the async loop? Can the exception handling be managed easily for all commands or do we need a test for each unique case?

Reply:
With the previous implementation of poll_status, indeed if we cancel a command and then try to get `command.result()` it will hang, but I have added an extra if statement in poll_status to check for status cancelled and if the status is cancelled to raise ApiError (do not know whether this error is appropriate).

For the second part I do not follow completely, but if an exception is to be raised from async operation, it will be raised once we try to get the result with result(), it will not be raised before that even if there is something wrong.

Reply:
Cool! That makes sense. Do we have a test for that as it's not clear if what you mentioned is covered in a unit test

Reply:
Yep, but I have used only one command for that,  the function is `test_get_file_cancelled`.

Reply:
That looks to cover it from the API side but it doesn't seem to account for the async case. I would expect a test that creates the command and returns a future. Then cancel the command and wait for the result